### PR TITLE
Feature friendship v1

### PR DESCRIPTION
**Pull request by mikethreels**
### in this milestone we:
- [x] implemented friendship associations.
- [x] added invite, accept and ignore buttons.
- [x] filtered the timeline posts on friends and user posts.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
+  <%unless (@confirmed_friends + @pending_friends).include?(@user.id) || @user.id == current_user.id%>
+    <%= button_to 'invite', invite_path, method: :post, params: { friend_id: @user.id } %>
+  <% end %>
```
Comment (maelfosso) [inline app/views/users/show.html.erb]: We called that logic into views.
Please, use **views helpers** to handle that.

Don't forget to handle it in other views like `index.html.erb`

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -3,6 +3,16 @@ class ApplicationController < ActionController::Base
...
 
+  def friend_search
+    all_friend_arr = current_user.friendships.pluck(:friend_id, :status),
+                     + current_user.reverse_friendships.pluck(:user_id, :status)
+    temporary_friends = []
+    all_friend_arr.select do |friend|
+      temporary_friends << friend[0] if yield(friend)
+    end
+    temporary_friends
+  end
```
Comment (maelfosso) [inline app/controllers/application_controller.rb]: It seems like there is a problem here

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle

Hello,

Thank you for the effort put into this milestone.
To move forward there are some changes to apply

1. After sign up, I got this error
Screenshot at 2020-07-24 23:55:12


2. Instead of using the `UsersController` for handling friendship request. I suggest you to create `Friendship` controllers and use the `create`, `delete` and `update` actions for everything related to friendship request

3. Write integrations tests.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by santiagorodriguezbermudez** [comment]
## Code review # 1

### On this code review we accomplished:
- [x] We created the friendship controller and helper methods
- [x] Move the logic from views into the helper methods
- [x] Implemented integration and unit tests

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team,  You have done a brilliant job with this project. :clap:
#### You can go ahead and merge !!!

**Pull request merged**